test/AutoDiff/control_flow.swift

```diff
@@ -690,8 +707,16 @@ ControlFlowTests.test("Loops") {
     }
     return outer
   }
+  // FIXME(TF-954): Investigate incorrect derivative related to addresses and
```

Comment:
I verified the correct derivative using PyTorch.
The reference PyTorch in TF-954 issue description. I decided not to paste it in the test source to save space.